django_squash/management/commands/lib/autodetector.py

```diff
@@ -54,6 +54,34 @@ def from_migration(cls, migration):
 class UniqueVariableName:
     def __init__(self):
         self.names = defaultdict(int)
+        self.functions = {}
+
+    def function(self, func):
+        if not callable(func):
+            raise ValueError('func must be a callable')
+
+        if isinstance(func, types.FunctionType) and func.__name__ == '<lambda>':
+            raise ValueError('func cannot be a lambda')
+
+        if inspect.ismethod(func) or inspect.signature(func).parameters.get('self') is not None:
+            raise ValueError('func cannot be part of an instance')
+
+        name = original_name = func.__qualname__
+        already_accounted = func in self.functions
+        if already_accounted:
+            return self.functions[func]
+
+        for i, _ in enumerate(itertools.count(), 2):
```

Comment:
This loop is mysterious. Please add a comment explaining in plain English what it does.

Reply:
LOL. Its an endless loop. that starts at 2 :P -- will do

Reply:
Done.